customer-config/checkout-config/che.plulist.v2.json

```diff
@@ -0,0 +1,221 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://raw.githubusercontent.com/extenda/hiiretail-json-schema-registry/master/customer-config/checkout-config/che.plulist.v1.json",
+  "title": "PLU List messages",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "required": [
+      "id",
+      "plu"
+    ],
+    "properties": {
+      "effectiveDate": {
+        "description": "The date when this PLU list should be applied, if present.",
+        "type": "string",
+        "format": "date",
+        "example": "2021-01-01"
+      },
+      "id": {
+        "description": "A unique identifier for the PLU list.",
+        "type": "string",
+        "minLength": 1,
+        "maxLength": 20,
+        "example": "top-5-items-plu"
+      },
+      "name": {
+        "description": "The PLU List name",
+        "type": "string",
+        "minLength": 1,
+        "maxLength": 40,
+        "example": "PLU List"
+      },
+      "description": {
+        "description": "The PLU List description",
+        "type": "string",
+        "minLength": 1,
+        "maxLength": 255,
+        "example": "The description for PLU List"
+      },
+      "deviceTypes": {
+        "description": "A list of device types that this PLUList applies to",
+        "type": "array",
+        "example": "[{'id': 'SelfCheckout'}]",
+        "items": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string",
+              "minLength": 1,
+              "maxLength": 40
+            }
+          }
+          }
+      },
+      "isModifiable": {
+        "description": "If it is possible to modify the content of this PLUList further down the hierarchy",
+        "type": "boolean",
+        "example": "true"
+      },
+      "isRemovable": {
+        "description": "If it is possible to remove the content of this PLUList further down the hierarchy",
+        "type": "boolean",
+        "example": "true"
+      },
+      "content": {
+        "description": "PLU lists can contain other PLU lists and/or items.",
+        "type": "array",
+        "items": {
+          "$ref": "#/$defs/PluContentType"
+        }
+      },
+      "additionalProperties": {
+        "description": "Additional custom properties can be added via this field.",
+        "type": "array",
+        "items": {
+          "$ref": "#/$defs/AdditionalProperty"
+        }
+      }
+    }
+  },
+  "$defs": {
+    "PluContentType": {
+      "type": "object",
+      "oneOf": [
+        {
+          "$ref": "#/$defs/PluItemType"
+        },
+        {
+          "$ref": "#/$defs/PluGroupType"
+        }
+      ]
+    },
+    "PluItemType": {
+      "description": "This is an entry for a specific item, and is placed inside a PLU list.",
+      "type": "object",
+      "required": [
+        "identifier",
+        "images",
+        "type"
+      ],
+      "properties": {
```

Comment:
We also need an `id` property for the `PluItemType`, so that CCC can apply PATCH operation to it. Otherwise, we'll not be able to merge this field.

The `id` field will simply duplicate the `identifier` field and can be ignored by the consumers

Reply:
This is confusing. Why not just use the identifier?

Reply:
We need both, otherwise, CCC API wouldn't be able to PATCH PLU lists. Everyone except for the frontend can ignore it.
FYI @mdovhopo

Reply:
You should patch using the identifier

Reply:
@msilfver this is pretty simple. we decided to use `id` as a merging key.

check out our adr - https://github.com/extenda/engineering-cloud-core-common/blob/master/docs/customer-controlled-configuration/internal/adr/0008-add-patch-operation-for-all-targets.md

This is a limitation that we agreed to have to simplify delivery on our side and make sure we deliver plu on time.

tldr: we need some field to use as merging key, so we can add elements to array easily. Since you do not want to use `id` instead of `identifier` as we asked, we though of compromise with @DmitriyStadnik:
* We will keep `identifier` field for you, so you do not have to change anything.
* We will have `id` in every element of any array, achieving unified name. this will be used by frontend as unique anchor.
* You will not need to use `id` if you do not want
* `id` will be always === `identifier`, to keep compatibility

Reply:
@msilfver Is there any reason why we can't use object instead of array on top level?

It would really simplify logic in CCC if it was object.

Current approach with array and `id` field:

```js
[
  { id: "plu1", content: { ... } },
  { id: "plu2", content: { ... } },
]
```

Object on top level where `id` field just become object keys:

```js
{
  plu1: { ... },
  plu2: { ... },
}
```

Reply:
i guess that would lead to a lot of changes on SD side... they want to do as little changes as possible...

Reply:
@glebbash No, it was just as it was before and as I assume most other schemas would look like.

Reply:
as a compromise we offer having both fields. But i (and FE devs) would prefer to have only `id` for consistency

Reply:
Both fields is even worse

Reply:
How about use object instead of array of objects and then you could keep only identifier but would also use it as object's key

Reply:
Clarified the property name to itemIdentifier

Reply:
@glebbash Changed to object